src/spdx/parser/tagvalue/parser/helper_methods.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2023 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import Optional, Callable, Any, Dict
+
+from spdx.model.checksum import Checksum, ChecksumAlgorithm
+from spdx.parser.error import SPDXParsingError
+from spdx.parser.logger import Logger
+from spdx.parser.parsing_functions import construct_or_raise_parsing_error
+
+
+def grammar_rule(doc):
+    # this is a helper method to use decorators for the parsing methods instead of docstrings
+    def decorate(func):
+        func.__doc__ = doc
+        return func
+
+    return decorate
+
+
+def str_from_text(text: Optional[str]) -> Optional[str]:
+    regex = re.compile("<text>((.|\n)+)</text>", re.UNICODE)
+    match = regex.match(text)
+    if match:
+        return match.group(1)
+    elif isinstance(text, str):
+        return text
+    else:
+        return None
+
+
+def parse_checksum(logger: Logger, checksum_str: str, line_number: int) -> Optional[Checksum]:
+    try:
+        algorithm, value = checksum_str.split(":")
+    except ValueError:
+        logger.append(
+            f"Couldn't split value for checksum in algorithm and value. Line: {line_number}")
+        return None
+    algorithm = ChecksumAlgorithm[algorithm.upper().replace("-", "_")]
+    value = value.strip()
+    try:
+        checksum = construct_or_raise_parsing_error(Checksum, {"algorithm": algorithm, "value": value})
+    except SPDXParsingError as err:
+        logger.append(err.get_messages())
+        checksum = None
+    return checksum
+
+
+def set_value(parsed_value: Any, dict_to_fill: Dict[str, Any], argument_name: Optional[str] = None,
+              method_to_apply: Callable = lambda x: x):
+    if not argument_name:
+        argument_name = str(parsed_value.slice[0])
```

Comment:
While correcting this, I was wondering if it would be better to not access the field `slice` and use the parameter `argument_name` as a mandatory parameter. Accessing a field from the class `YaccProduction` which has no dediacted method seems a bit hacky and I think this is not really intended. Additionally, we create a dependency between the name of the grammar rule and the value in the data model which I think is a bit of an unwanted behaviour. On the other hand we would have more strings (and potential error sources) when writing each argument name individually and couldn't combine the parsing mtehods for generic values. Probably the last one is a string argument for the current implementation. What do you think?